python/ray/air/integrations/comet.py

```diff
@@ -1,6 +1,10 @@
 import os
 from typing import Dict, List
 
+import pyarrow.fs
+
+from ray.air.checkpoint import Checkpoint as LegacyCheckpoint
+from ray.train._checkpoint import Checkpoint as Checkpoint
```

Comment:
We can use `ray.train.Checkpoint` directly, then keep the code gated w/ the env var:

```python
from ray.train import Checkpoint

...

if _use_storage_path():
    trial.checkpoint.path
else:
    trial.checkpoint.dir_or_data
```